src/application.c

```diff
@@ -191,6 +191,10 @@ static void save_state_to_file_or_folder(struct swappy_state *state,
   }
 
   g_object_unref(pixbuf);
+
+  if (state->config->quit_after_export) {
+      exit(0);
```

Comment:
This is too harsh and will not release memory properly. Better find a more gtk friendly way, or add a new state.

Reply:
I introduced a new function early_exit that should gracefully kill the application.